docs/Music Macro Language.md

```diff
@@ -0,0 +1,989 @@
+
+**Example:**
+```
+scriptctr 0  # Sets the script counter to 0.  This has no impact on the sequence itself.
+```
+
+**halt**
+
+**Description:** Halts any further processing of the current sequence after this tick.
+
+**Parameters:** None.
+
+**Example:**
+```
+ldchan 1, chan1
+halt              # Channel 1 will run for one tick before the script ends.
+```
+
+**stseq** *val*, *offset/label*
+
+**Description:** Writes the total of `$V` + `val` to the offset within the sequence script in its binary form.  The easiest way to use this command is to use a label within the script pointing to the command/data you wish to modify.
+
+Note that this instruction modifies the sequence script itself and is a form of self-modifying code.  This can be leveraged to make highly dynamic sequences (and is used to great effect in the sound effects sequence), but using it improperly can easily cause sequence corruption and undefined behavior.
+
+**Parameters:**
+* `val` [integer: 0-255] - Constant to add to `$V`.
+* `offset/label` [integer: 0-32767 or any label] - Offset or label to location in sequence to be modified.
+
+**Example:**
+```
+ldchan 5, chan5
+ldi 7
+stseq 0, chan5_instr+1   # Changes the instrument specified in chan5 to 7.
+delay 500
+end
+
+.channel chan5
+short
+.channel chan5_instr
+instr 3
+...
+```
+
+**sub** *val*
+
+**Description:** Subtracts the specified integer from the current value stored in the `$V` register.
+
+**Parameters:**
+* `val` [integer: 0-255] - Value to subtract
+
+**Example:**
+```
+ldio 3   # Load value from IO port 3 into $V
+sub 2    # $V -= 2
+stio 3   # Store $V to IO port 3
+```
+
+**and** *val*
+
+**Description:** Performs a bitwise and operation between the value stored in the `$V` register and the value specified.
+
+**Parameters:**
+* `val` [integer: 0-255] - Value to perform a bitwise and with
+
+**Example:**
+```
+ldio 3     # Load value from IO port 3 into $V
+and 0xF0   # Set lower four bits to 0
+stio 2     # Store resulting value into IO port 2
+```
+
+**ldi** *val*
+
+**Description:** Sets the `$V` register to the provided integer.
+
+**Parameters:**
+* `val` [integer: 0-255] - Value to load into the register
+
+**Example:**
+```
+ldi 2    # $V = 2
+stio 0   # Set IO port 0 to 2
+```
+
+**dyncall** *table*
+
+**Description:** Makes a function call using an entry from the specified table in a data section.  The value in the `$V` register is treated as the index to look up within the table.  Once an `end` command is reached in the called function, the player will return back and execute the next instruction after this one.
+
+If `$V` is currently set to -1 or the call depth is at 3, then the function call will fail and be skipped.  Note that the sequence player does not perform any bounds checking on `$V`; if it's larger than the table, the sequence player will jump to a potentially unexpected location in the script and could result in undefined behavior.
```

Comment:
What does the `dyn` mean here? I suppose it's tied to all the `dynTable` commands to indicate they're all in a group? (noting decomp does also use the `dyn` terminology)

Reply:
TBD.